[REF] beesdoo_shift -> shift, beesdoo_worker_status -> shift_worker_status, beesdoo_shift_attendance -> shift_attendance

robinkeunen opened the pull request:
## Description

fix https://github.com/beescoop/Obeesdoo/issues/361
fix #343 
fix #362 

## Checklist before approval

- [ ] Tests are present (or not needed).
- [ ] Credits/copyright have been changed correctly.
- [ ] Change log snippet is present.
- [x] (If a new module) Moving this to OCA has been considered.

## Checklist after merging

- [ ] Adjust and merge https://github.com/coopiteasy/cie-custom/pull/93

robinkeunen committed:
[REF] Rename module: beesdoo_shift -> shift

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

carmenbianca committed:
[REF] shift: beesdoo_shift -> shift

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

carmenbianca committed:
[REF] shift: beesdoo.shift -> shift

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

carmenbianca committed:
[REF] shift: beesddoo.shift -> shift

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

carmenbianca committed:
[REF] shift: beesddoo_shift -> shift

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

carmenbianca committed:
[FIX] shift: Manually remove some mentions of BEES in code

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

carmenbianca committed:
[REM] shift: Migration scripts

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

carmenbianca committed:
[FIX] shift: Add OCA to authors

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

carmenbianca committed:
[IMP] shift: Add pre-init migration hook

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

carmenbianca committed:
[REF] Rename module: beesdoo_shift_attendance -> shift_attendance

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

carmenbianca committed:
[REF] shift_attendance: beesdoo_shift -> shift

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

carmenbianca committed:
[REF] shift_attendance: beesdoo.shift -> shift

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

carmenbianca committed:
[REF] shift_attendance: Manually remove beesdoo references

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

carmenbianca committed:
[REF] Rename module: beesdoo_worker_status -> shift_worker_status

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

carmenbianca committed:
[FIX] shift_worker_status: Add OCA to authors

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

carmenbianca committed:
[ADD] shift_worker_status: Pre-init hook

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

carmenbianca committed:
[FIX] shift_attendance: Add OCA to authors

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

carmenbianca committed:
[ADD] shift_attendance: Pre-init hook

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

carmenbianca commented:
Closed #491 and #492 to put that in this PR.

carmenbianca committed:
[REF] shift_worker_status: beesdoo_shift -> shift

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

carmenbianca committed:
[REF] shift_worker_status: beesdoo.shift -> shift

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

carmenbianca committed:
[REF] shift_worker_status: beesdoo_worker_status -> shift_worker_status

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

carmenbianca committed:
[FIX] shift_worker_status: Manually remove beesdoo references

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

carmenbianca committed:
[REF] shift_worker_status: task -> shift_shift

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

carmenbianca committed:
[FIX] Satisfy pre-commit

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

carmenbianca committed:
[REF] beesdoo_worker_status_shift_swap: beesdoo_worker_status -> shift_worker_status

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

carmenbianca committed:
[REF] beesdoo.shift -> shift

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

carmenbianca committed:
[REF] beesdoo_shift -> shift

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

carmenbianca committed:
[FIX] shift: Fix translations

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

carmenbianca committed:
[FIX] shift_attendance: Fix translations

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

carmenbianca commented:
Copying a comment from Mattermost:

There's a bug in the renaming of beesdoo_shift to shift. There's a workaround, but I'm making a note of it here in case someone wants to tackle this in the new year.

---

There's a bug where Odoo tries to create a duplicate entry in the `ir_config_parameter` table when installing the new modules' data, breaking the `ir_config_parameter_key_uniq` restraint. For some reason this bug is only triggered on bees. I think it's because bees is the only database where the ir.config_parameter values differ from their defaults, but this is hard to verify.

The current workaround is like so:

before update:

```sql
-- remember the key-value pairs.
select key,value from ir_config_parameter where key in ('min_percentage_presence', 'regular_next_shift_limit');
-- remove them.
delete from ir_config_parameter where key in ('regular_next_shift_limit', 'min_percentage_presence');
```

run update

keep odoo turned off

do some sql:

```sql
-- make sure the values are correct.
update ir_config_parameter set value = 10 where key = 'regular_next_shift_limit';
update ir_config_parameter set value = 40 where key = 'min_percentage_presence';
```

turn odoo on.

---

I am convinced that there must exist a better way. Perhaps:

- in the pre-init hook, read the key-value pairs of relevant parameters, and store them in a file in `/tmp`.
- Subsequently in the pre-init hook, delete the records from the DB.
- Module initialisation happens. The parameters are re-created.
- In a post-init hook, read the file in `/tmp` and update the values of the (potentially renamed) keys.

This seems quite flimsy, though. Is it worth the extra effort to circumvent a one-time workaround during MeP?

---

Tagging @victor-champonnois because it's adjacent to work you're also doing.

huguesdk committed:
[FIX] fix beesdoo renaming scripts

* rename many2many tables and their columns.
* rename constraints and indexes.

carmenbianca commented on shift/rename_beesdoo.py:
```diff
@@ -11,23 +11,138 @@
+    ("beesdoo.shift.type", "shift.type"),
+]
+TABLES_TO_RENAME = [
+    ("beesddoo_shift_assign_super_coop", "shift_assign_super_coop"),
+    ("beesddoo_shift_generate_planning", "shift_generate_planning"),
+    (
+        "beesddoo_shift_generate_shift_template_line",
+        "shift_generate_shift_template_line",
+    ),
+    ("beesddoo_shift_generate_shift_template", "shift_generate_shift_template"),
+    ("beesdoo_shift_daynumber", "shift_daynumber"),
+    ("beesdoo_shift_extension", "shift_extension"),
+    ("beesdoo_shift_holiday", "shift_holiday"),
+    ("beesdoo_shift_journal", "shift_journal"),
+    ("beesdoo_shift_planning", "shift_planning"),
+    ("beesdoo_shift_shift", "shift_shift"),
+    ("beesdoo_shift_subscribe", "shift_subscribe"),
+    ("beesdoo_shift_template", "shift_template"),
+    ("beesdoo_shift_temporary_exemption", "shift_temporary_exemption"),
+    ("beesdoo_shift_type", "shift_type"),
```
This violates DRY. I'd prefer to generate this half of the list from MODELS_TO_RENAME. Thoughts?

carmenbianca commented on shift/rename_beesdoo.py:
```diff
@@ -11,23 +11,138 @@
+        ),
+    ],
 }
+INDEXES_TO_RENAME = [
+    (
+        "beesddoo_shift_assign_super_c_beesddoo_shift_assign_super_c_idx",
+        "shift_assign_super_coop_shift_sh_shift_assign_super_coop_id_idx",
+    ),
+    (
+        "beesddoo_shift_assign_super_coop_bee_beesdoo_shift_shift_id_idx",
+        "shift_assign_super_coop_shift_shift_rel_shift_shift_id_idx",
+    ),
+    (
+        "beesdoo_shift_journal_cooperative__beesdoo_shift_journal_id_idx",
+        "cooperative_status_shift_journal_rel_shift_journal_id_idx",
+    ),
+    (
+        "beesdoo_shift_journal_cooperative_sta_cooperative_status_id_idx",
+        "cooperative_status_shift_journal_rel_cooperative_status_id_idx",
+    ),
+    (
+        "beesdoo_shift_template_res_partne_beesdoo_shift_template_id_idx",
+        "res_partner_shift_template_rel_shift_template_id_idx",
+    ),
+]
```
The naming schemes appear very different here. How is this possible?

Also, are you sure this doesn't contain typos? There's a `res_partne`, a double `__`, a `cooperative_sta`, etc etc etc.

huguesdk committed:
[IMP] add comments to beesdoo_shift rename script

robinkeunen commented on shift/models/task.py:
```diff
@@ -8,7 +8,7 @@
 
 
 class Task(models.Model):
```
Can you rename the class to ShiftShift and the file to shift_shift ? Or add a todo for after Monday's release.

robinkeunen commented on shift/__manifest__.py:
```diff
@@ -0,0 +1,47 @@
+        "demo/templates.xml",
+    ],
+    "license": "AGPL-3",
+    "pre_init_hook": "rename_beesdoo",
```
👍

robinkeunen commented on shift/models/planning.py:
```diff
@@ -187,24 +187,23 @@ def get_future_shifts(
+    planning_id = fields.Many2one("shift.planning", required=True)
+    day_nb_id = fields.Many2one("shift.daynumber", string="Day", required=True)
+    task_type_id = fields.Many2one("shift.type", string="Type")
+    # FIXME removed because shift.sheet is from another module.
+    # attendance_sheet_id = fields.Many2one('shift.sheet', string="Attendance Sheet")  # noqa
```
These comments can be removed.

robinkeunen commented on shift/rename_beesdoo.py:
```diff
@@ -0,0 +1,220 @@
+        ("beesdoo_shift_journal_id", "shift_journal_id")
+    ],
+}
+# constraints and indexes are normally correctly renamed by the openupgrade
+# functions, but these fail for identifiers that have been truncated to the
+# maximum length (63 characters). they are thus manually renamed here. note
+# that these are cosmetic changes, not doing them would not prevent the module
+# from working correctly.
```
👏

robinkeunen commented on shift/rename_beesdoo.py:
```diff
@@ -0,0 +1,220 @@
+    "cooperative_status_shift_journal_rel": [
+        (
+            "beesdoo_shift_journal_coopera_beesdoo_shift_journal_id_coop_key",
+            "cooperative_status_shift_jour_shift_journal_id_cooperative__key",
```
Is the double underscore intentional ?

carmenbianca replied:
yes

robinkeunen commented on shift/rename_beesdoo.py:
```diff
@@ -0,0 +1,220 @@
+            "shift_assign_super_coop_shift_shift_assign_super_coop_id_sh_key",
+        ),
+        (
+            "beesddoo_shift_assign_super_c_beesddoo_shift_assign_super__fkey",
```
Same here.

robinkeunen commented on shift/rename_beesdoo.py:
```diff
@@ -0,0 +1,220 @@
+            OLD_MODULE_NAME, NEW_MODULE_NAME
+        )
+    )
+    # this query comes from openupgrade.update_module_names(), which cannot be
+    # used directly here because the module with the old name still exists.
+    openupgrade.logged_query(
```
👍

robinkeunen commented on shift/wizard/assign_super_coop.py:
```diff
@@ -2,8 +2,8 @@
 
 
 class AssignSuperCoop(models.TransientModel):
-    _name = "beesddoo.shift.assign_super_coop"
```
🙄 at the previous name ...

In the new name, did you consider replacing underscores by dots ?

carmenbianca replied:
Not replacing the underscores with dots this late in the PR. It's probably fine.

robinkeunen commented on shift/wizard/assign_super_coop.xml:
```diff
@@ -1,7 +1,7 @@
 <odoo>
     <record model="ir.ui.view" id="assign_super_coop_view_form">
         <field name="name">Assign Super Coop</field>
-        <field name="model">beesddoo.shift.assign_super_coop</field>
```
Damn, that double `d` hurts the eye.

robinkeunen commented on shift/wizard/instanciate_planning.py:
```diff
@@ -2,15 +2,15 @@
 
 
 class InstanciatePlanning(models.TransientModel):
-    _name = "beesddoo.shift.generate_planning"
```
same :rage4: for old name, same question about `_`.

robinkeunen commented on shift_attendance/tests/test_beesdoo_shift.py:
```diff
@@ -8,54 +8,44 @@
 from odoo.tests.common import TransactionCase
 
 
-class TestBeesdooShift(TransactionCase):
```
Maybe ?
```py
class TestShift(TransactionCase):
```

robinkeunen commented on shift_worker_status/i18n/beesdoo_worker_status.pot:
```diff
@@ -0,0 +1,115 @@
+# Translation of Odoo Server.
+# This file contains the translation of the following modules:
```
File name is off.

robinkeunen commented on shift_worker_status/models/shift_shift.py:
```diff
@@ -3,8 +3,8 @@
 
-class Task(models.Model):
-    _inherit = "beesdoo.shift.shift"
+class Shift(models.Model):
```
`ShiftShift` for consistency ?

robinkeunen commented on shift_worker_status/tests/test_beesdoo_shift.py:
```diff
@@ -7,39 +7,35 @@
 from odoo.tests.common import TransactionCase
 
 
-class TestBeesdooShift(TransactionCase):
```
`TestShiftWorkerStatus`

robinkeunen reviewed (commented):
@carmenbianca pfiou that was a long one. I can't approve / reject because I'm the owner of the PR. Appart from a translation file which kept the _beesdoo_ name, the rest seems good. 

The few renaming suggestions can be left to after the release to avoid disrupting it.

carmenbianca committed:
[FIX] shift: Rename config parameters, copy old values

Previously, when the parameters were not renamed, Odoo would write the
default values to the parameters during init.

Now, we create new parameters, get the old values, write them to the new
parameters, then delete the old parameters. There probably exists a
better way, but this'll do for now.

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

carmenbianca committed:
[FIX] shift_worker_status: Copy old value from parameter

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

carmenbianca committed:
[FIX] shift_attendance: Copy old values from parameters

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

carmenbianca committed:
[FIX] shift: Prefix parameters in code

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

carmenbianca committed:
[IMP] Add logging entry

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

carmenbianca committed:
[FIX] Actually preserve old param values

There are two fixes in this commit:

- Use .search() instead of .get_param() for a more robust search of the
  old value.
- Rename all XMLIDs. When they XMLIDs aren't renamed, the values get
  overridden during init, even though the config_parameter keys were
  renamed.

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

carmenbianca committed:
[FIX] shift: Improve comment

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

carmenbianca committed:
[FIX] shift: rename classes, files

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

carmenbianca committed:
[FIX] shift: Remove unnecessary comment

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

carmenbianca committed:
[FIX] shift: "instanciate" typo

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

carmenbianca committed:
[FIX] shift_attendance: Rename file

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

carmenbianca committed:
[FIX] shift_worker_status: Rename file

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

carmenbianca committed:
[FIX] shift_worker_status: rename class

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

carmenbianca committed:
[FIX] shift_worker_status: rename test class

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

carmenbianca committed:
[FIX] Run pre-commit

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

carmenbianca committed:
[FIX] shift_worker_status: rename test file

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

remytms commented:
We should not forget to merge this linked PR, after this one: https://github.com/coopiteasy/cie-custom/pull/93

huguesdk commented on beesdoo_shift/__manifest__.py:
```diff
@@ -5,37 +5,13 @@
 {
     "name": "BEES coop Shift Management",
-    "summary": """Generate and manage shifts for cooperators.""",
+    "summary": """Emptied.""",
```
instead of "Emptied" as module summary, why not use something more explicit like: "Obsolete (replaced by the shift module)", with the readme saying that the module can be safely uninstalled? (same for the 2 other affected modules.)

huguesdk commented on beesdoo_shift/static/description/icon.png:
this file should be moved to the `shift` module instead of being deleted. it is referenced by `shift/views/menu.xml`.

huguesdk commented on beesdoo_shift_swap/models/shift_swap.py:
```diff
@@ -101,7 +101,7 @@ def _subscribe_new_shift_if_generated(self):
     def update_shift_data(self, shift, swap_subscription_done):
         """
-        See method info in model beesdoo.shift.swap.mixin
+        See method info in model shift.swap.mixin
```
this line should stay like it was, as `beesdoo.shift.swap.mixin` has not been renamed.

huguesdk commented on shift/__manifest__.py:
```diff
@@ -0,0 +1,48 @@
+        Elouan Le Bars,
+        Coop IT Easy SC,
+        Odoo Community Association (OCA),
+        """,
```
```suggestion
    "author": (
        "Thibault Francois, "
        "Elouan Le Bars, "
        "Coop IT Easy SC, "
        "Odoo Community Association (OCA)"
    ),
```

this should not be a multiline string, as it results in this:
```python
"\n        Thibault Francois,\n        Elouan Le Bars,\n        Coop IT Easy SC,\n        Odoo Community Association (OCA),\n        "
```

same comment for the other 2 modules.

huguesdk commented on shift/models/cooperative_status.py:
```diff
@@ -105,7 +105,7 @@ def _get_status(self):
     )
     unsubscribed = fields.Boolean(default=False, help="Manually unsubscribed")
-    resigning = fields.Boolean(default=False, help="Want to leave the beescoop")
+    resigning = fields.Boolean(default=False, help="Wants to leave the cooperative")
```
nice catch! 👍

huguesdk commented on shift/views/task.xml:
rename this file to `shift.xml` as well as the `task_*` xmlids?

same comment for `task_template.xml`.

carmenbianca replied:
Yes. I propose to delay this for another PR.

huguesdk commented on shift/__manifest__.py:
```diff
@@ -0,0 +1,48 @@
+        """,
+    "website": "https://github.com/beescoop/Obeesdoo",
+    "category": "Cooperative management",
+    "version": "12.0.4.0.0",
```
the version of the modules have not been changed. should this pull request be merged as major, or should they start back at 12.0.1.0.0 since they are new modules?

huguesdk reviewed (changes requested):
impressive work! it’s crazy how many times and in how many places the module and the model names are present!

it would be better to squash all (or most) commits, not only because having the full history is not so useful in this case, but especially because there is a loss of history here: the first commit removes `beesdoo_shift`, which is re-added as `shift` later. in this case, git will not be able to link the history between the moved files. the moving should be done in the same commit.

impressive work also in all the modules that depend on `beesdoo_shift`, especially in `beesdoo_shift_swap`!

carmenbianca committed:
[IMP] shift: Re-add icon.png

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

carmenbianca committed:
[FIX] beesdoo_shift_swap: Fix comment

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

carmenbianca committed:
[FIX] Give instructions to uninstall modules

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

carmenbianca committed:
[FIX] author fields in manifests

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

carmenbianca committed:
[UPD] bump versions

Signed-off-by: Carmen Bianca BAKKER <carmen@coopiteasy.be>

carmenbianca commented:
resolved your review @huguesdk 

also rebased to no longer lose history with files renaming/being deleted/whatever.

will forward-port my changes to #504

huguesdk reviewed (approved):
thanks @carmenbianca! 👍

huguesdk committed:
[REL] increment versions of dependent modules

increment versions of dependent modules after renaming of beesdoo_shift
and beesdoo_worker_status.

huguesdk commented:
/ocabot merge nobump

carmenbianca commented:
thank you @huguesdk !
